R/expr-string.R

```diff
@@ -0,0 +1,1136 @@
+#' If no fractional second component is found then the default is `"us"` (microsecond).
+#' @param dtype The data type to convert into. Can be either `pl$Date`,
+#' [`pl$Datetime()`][DataType_Datetime], or `pl$Time`.
+#' @param format Format to use for conversion. Refer to
+#' [the chrono crate documentation](https://docs.rs/chrono/latest/chrono/format/strftime/index.html)
+#' for the full specification. Example: `"%Y-%m-%d %H:%M:%S"`.
+#' If `NULL` (default), the format is inferred from the data.
+#' Notice that time zone `%Z` is not supported and will just ignore timezones.
+#' Numeric time zones like `%z` or `%:z` are supported.
+#' @param ... Not used.
+#' @param strict If `TRUE` (default), raise an error if a single string cannot
+#' be parsed. If `FALSE`, produce a polars `null`.
+#' @param exact If `TRUE` (default), require an exact format match. If `FALSE`,
+#' allow the format to match anywhere in the target string.
+#' Conversion to the Time type is always exact.
+#' Note that using `exact = FALSE` introduces a performance penalty -
+#' cleaning your data beforehand will almost certainly be more performant.
+#' @param cache Use a cache of unique, converted dates to apply the datetime
+#' conversion.
+#' @inheritParams ExprDT_replace_time_zone
+#' @return [Expr][Expr_class] of Date, Datetime or Time type
+#' @seealso
+#' - [`<Expr>$str$to_date()`][expr_str_to_date]
+#' - [`<Expr>$str$to_datetime()`][expr_str_to_datetime]
+#' - [`<Expr>$str$to_time()`][expr_str_to_time]
+#' @examples
+#' # Dealing with a consistent format
+#' s <- as_polars_series(c("2020-01-01 01:00Z", "2020-01-01 02:00Z"))
+#'
+#' s$str$strptime(pl$Datetime(), "%Y-%m-%d %H:%M%#z")
+#'
+#' # Auto infer format
+#' s$str$strptime(pl$Datetime())
+#'
+#' # Datetime with timezone is interpreted as UTC timezone
+#' as_polars_series("2020-01-01T01:00:00+09:00")$str$strptime(pl$Datetime())
+#'
+#' # Dealing with different formats.
+#' s <- as_polars_series(
+#'   c(
+#'     "2021-04-22",
+#'     "2022-01-04 00:00:00",
+#'     "01/31/22",
+#'     "Sun Jul  8 00:34:60 2001"
+#'   ),
+#'   "date"
+#' )
+#'
+#' s$to_frame()$select(
+#'   pl$coalesce(
+#'     pl$col("date")$str$strptime(pl$Date, "%F", strict = FALSE),
+#'     pl$col("date")$str$strptime(pl$Date, "%F %T", strict = FALSE),
+#'     pl$col("date")$str$strptime(pl$Date, "%D", strict = FALSE),
+#'     pl$col("date")$str$strptime(pl$Date, "%c", strict = FALSE)
+#'   )
+#' )
+#'
+#' # Ignore invalid time
+#' s <- as_polars_series(
+#'   c(
+#'     "2023-01-01 11:22:33 -0100",
+#'     "2023-01-01 11:22:33 +0300",
+#'     "invalid time"
+#'   )
+#' )
+#'
+#' s$str$strptime(
+#'   pl$Datetime("ns"),
+#'   format = "%Y-%m-%d %H:%M:%S %z",
+#'   strict = FALSE
+#' )
+expr_str_strptime <- function(
+    dtype,
+    format = NULL,
+    ...,
+    strict = TRUE,
+    exact = TRUE,
+    cache = TRUE,
+    ambiguous = "raise") {
+  if (!is_polars_dtype(dtype)) {
+    abort("`dtype` must be of class `RPolarsDataType`.")
+  }
```

Comment:
Please use `check_polars_dtype()` as like this:
https://github.com/eitsupi/neo-r-polars/blob/2672fce59b94c67655ac0ab7a13a1995db4da437/R/expr-expr.R#L265-L272

And, please check dots by `check_dots_empty0()`